scripts/manual_operations/manual_submission.py

```diff
@@ -165,36 +197,29 @@ def main():
             sys.exit(1)
         run_numbers = range(args.start_run_number, args.e + 1)
 
-    print("Logging into ICAT")
-    icat_client = ICATClient()
-    try:
-        icat_client.connect()
-    except ICATSessionError:
-        print("Couldn't connect to ICAT. Continuing without ICAT connection.")
-        icat_client = None
+    instrument = args.instrument.upper()
 
-    print("Logging into Database")
-    database_client = DatabaseClient()
-    try:
-        database_client.connect()
-    except ConnectionException:
-        print("Couldn't connect to Database. Continuing without Database connection.")
-        database_client = None
+    return run_numbers, instrument
 
-    print("Logging into ActiveMQ")
-    activemq_client = QueueClient()
-    try:
-        activemq_client.connect()
-    except (ConnectionException, ValueError):
-        print("Couldn't connect to ActiveMQ. Continuing without ActiveMQ connection.")
-        activemq_client = None
 
-    instrument = args.instrument.upper()
+def main():
+    """
+    Collate user input and initialise clients before attempting to submit new runs
+    """
+    run_numbers, instrument = handle_input()
+    icat_client = login_icat()
+    database_client = login_database()
+    if not icat_client and not database_client:
+        raise RuntimeError("Unable to proceed. Unable to connect to ICAT or Database. "
+                           "At least one connection is required to perform manual submission.")
+
+    activemq_client = login_queue()
 
     for run in run_numbers:
         location, rb_num = get_location_and_rb(database_client, icat_client, instrument, run, "nxs")
-        submit_run(activemq_client, rb_num, instrument, location, run)
+        if location and rb_num:
+            submit_run(activemq_client, rb_num, instrument, location, run)
```

Comment:
Perhaps an `else: print()` letting the user know run with location _x_ and rb_num _y_ wasn't found?